pkgs/tools/filesystems/dwarfs/default.nix

```
@@ -0,0 +1,104 @@
+{ stdenv
+, substituteAll
+, fetchFromGitHub
+, lib
+
+, bison
+, boost
+, cmake
+, double-conversion
+, fmt_8
+, fuse3
+, gflags
+, glog
+, gtest
+, jemalloc
+, libarchive
+, libevent
+, libunwind
+, lz4
+, openssl
+, pkg-config
+, ronn
+, xxHash
+, xz
+, zstd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dwarfs";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "mhx";
+    repo = "dwarfs";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-bGJkgcq8JxueRTX08QpJv1A0O5wXbiIgUY7BrY0Ln/M=";
+  };
+
+  # taken from @tpwrules's derivation
+  patches = [
+    (substituteAll {
+      src = ./version_info.patch;
+
+      gitRev = "v${version}";
+      gitDesc = "v${version}";
+      gitBranch = "v${version}";
+      gitId = "v${version}"; # displayed as version number
+
+      versionMajor = builtins.elemAt (builtins.splitVersion version) 0;
+      versionMinor = builtins.elemAt (builtins.splitVersion version) 1;
+      versionPatch = builtins.elemAt (builtins.splitVersion version) 2;
+    })
+  ];
+
+  cmakeFlags = [
+    "-DPREFER_SYSTEM_ZSTD=ON"
+    "-DPREFER_SYSTEM_XXHASH=ON"
+    "-DPREFER_SYSTEM_GTEST=ON"
+
+    # may be added under an option in the future
+    # "-DWITH_LEGACY_FUSE=ON"
+    "-DWITH_TESTS=ON"
+  ];
+
+  nativeBuildInputs = [
+    bison
+    cmake
+    pkg-config
+    ronn
+  ];
+
+  buildInputs = [
+    # dwarfs
+    boost
+    fmt_8
+    fuse3
+    jemalloc
+    libarchive
+    lz4
+    zstd
+    xxHash
```

Did you want to keep them sorted alphabetically? (It doesn't really matter but it's very close so I think it was intentional...?)

```suggestion
    xxHash
    zstd
```